mockall/tests/automock_foreign_c.rs

```diff
@@ -37,3 +51,28 @@ fn returning() {
     ctx.expect().returning(i64::from);
     assert_eq!(42, unsafe{mock_ffi::foo(42)});
 }
+
+/// Ensure that the mock function can be called from C by casting it to a C
+/// function pointer.
+#[test]
+fn c_abi() {
+    let ctx = mock_ffi::foo_context();
+    ctx.expect().returning(i64::from);
+    let p: unsafe extern "C" fn(u32) -> i64 = mock_ffi::foo;
+    assert_eq!(42, unsafe{p(42)});
+}
+
+#[test]
+fn c_abi_with_qsort() {
```

Comment:
I don't think this qsort stuff adds any extra benefit.  All this test does it cast the mock function to a C function pointer, then call that.  But it's calling it from Rust code (the ABI of qsort doesn't matter).  And it doesn't depend on name mangling in any way.  I think you should just delete this test case.  Now, if there were _actual C code_ calling the mock function, that would be different.